drools-wb-screens/drools-wb-guided-dtree-editor/drools-wb-guided-dtree-editor-client/src/main/java/org/drools/workbench/screens/guided/dtree/client/widget/GuidedDecisionTreeWidget.java

```diff
@@ -117,296 +98,323 @@
             return;
         }
 
         final int scrollWidth = getElement().getScrollWidth();
         final int scrollHeight = getElement().getScrollHeight();
-        if ( cx > scrollWidth || cy > scrollHeight ) {
-            dropContext.setContext( null );
+        if (cx > scrollWidth || cy > scrollHeight) {
+            dropContext.setContext(null);
             return;
         }
 
         //Add the new Node to it's parent (unless this is the first node)
         final BaseGuidedDecisionTreeShape uiParent = dropContext.getContext();
-        boolean addShape = ( ( getShapesInCanvas().size() == 0 && ( uiChild instanceof TypeShape ) ) || ( getShapesInCanvas().size() > 0 && uiParent != null ) );
+        boolean addShape = ((getShapesInCanvas().size() == 0 && (uiChild instanceof TypeShape)) || (getShapesInCanvas().size() > 0 && uiParent != null));
         boolean addChildToParent = uiParent != null;
 
-        if ( addShape ) {
-            uiChild.setX( cx );
-            uiChild.setY( cy );
-
-            if ( addChildToParent ) {
-                uiParent.addChildNode( uiChild );
-                uiParent.getModelNode().addChild( uiChild.getModelNode() );
+        if (addShape) {
+            uiChild.setX(cx);
+            uiChild.setY(cy);
 
-            } else if ( uiChild instanceof TypeShape ) {
+            if (addChildToParent) {
+                uiParent.addChildNode(uiChild);
+                uiParent.getModelNode().addChild(uiChild.getModelNode());
+            } else if (uiChild instanceof TypeShape) {
                 uiRoot = uiChild;
-                model.setRoot( ( (TypeShape) uiChild ).getModelNode() );
+                model.setRoot(((TypeShape) uiChild).getModelNode());
             }
 
-            addShape( uiChild );
+            addShape(uiChild);
 
             //Notify other Panels of a Shape being added
-            shapeAddedEvent.fire( new ShapeAddedEvent( uiChild ) );
+            shapeAddedEvent.fire(new ShapeAddedEvent(uiChild));
         }
     }
 
-    private double getX( double xShapeEvent ) {
+    private double getX(double xShapeEvent) {
         return xShapeEvent - getAbsoluteLeft();
     }
 
-    private double getY( double yShapeEvent ) {
+    private double getY(double yShapeEvent) {
         return yShapeEvent - getAbsoluteTop();
     }
 
     @Override
     public void clear() {
-        if ( Window.confirm( GuidedDecisionTreeConstants.INSTANCE.confirmDeleteDecisionTree() ) ) {
+        if (Window.confirm(GuidedDecisionTreeConstants.INSTANCE.confirmDeleteDecisionTree())) {
             super.clear();
-            clearEvent.fire( new ClearEvent() );
+            clearEvent.fire(new ClearEvent());
             uiRoot = null;
         }
     }
 
     @Override
-    public void deleteShape( final WiresBaseShape shape ) {
-        if ( Window.confirm( GuidedDecisionTreeConstants.INSTANCE.confirmDeleteDecisionTreeNode() ) ) {
+    public void deleteShape(final WiresBaseShape shape) {
```

Comment:
The real changes are in this method.

I only call ```layout()``` if it was not the root node that was deleted.